controller/api/destination/server.go

```diff
@@ -353,6 +337,66 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
+// sendEndpointProfile will send a DestinationProfile response that contains an
+// endpoint to the client. It accepts the gRPC stream object an optional pod and
+// the port used in the GetProfile request. If the pod argument is supplied,
+// then sendEndpointProfile will reply with an endpoint; otherwise, the response
+// will contain the default profile.
+func (s *server) sendEndpointProfile(stream pb.Destination_GetProfileServer, pod *corev1.Pod, port uint32) error {
+	log := s.log
+	var endpoint *pb.WeightedAddr
+	opaquePorts := make(map[uint32]struct{})
+	var err error
+	if pod != nil {
+		podSet := podToAddressSet(s.k8sAPI, pod).WithPort(port)
+		podID := watcher.PodID{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+		}
+		var ok bool
+		opaquePorts, ok, err = getPodOpaquePortsAnnotations(pod)
+		if err != nil {
+			log.Errorf("failed to get opaque ports annotation for pod: %s", err)
+		}
+
+		// If the opaque ports annotation was not set, then set the
+		// endpoint's opaque ports to the default value.
+		if !ok {
+			opaquePorts = s.defaultOpaquePorts
+		}
+
+		skippedInboundPorts, err := getPodSkippedInboundPortsAnnotations(pod)
+		if err != nil {
+			log.Errorf("failed to get ignored inbound ports annotation for pod: %s", err)
+		}
+
+		endpoint, err = toWeightedAddr(podSet.Addresses[podID], opaquePorts, skippedInboundPorts, s.enableH2Upgrade, s.identityTrustDomain, s.controllerNS, s.log)
+		if err != nil {
+			return err
+		}
+		// `Get` doesn't include the namespace in the per-endpoint
+		// metadata, so it needs to be special-cased.
+		endpoint.MetricLabels["namespace"] = pod.Namespace
+	}
+
+	// When the IP does not map to a service, the default profile is sent
+	// without subscribing for future updates. If the IP/DNS name is mapped to a
+	// pod, then the endpoint will be set in the response.
```

Comment:
Since this function is only called when dealing with a pod or a hostname we can update this comment. We've known at this point that the IP does not map to a service.